### DISCUSSION TITLE
app directory 404 page

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#45597] [Help]
hello, in pages dir, we can make a 404.jsx file and it will be website 404 page. how can I intorduce 404 page in app dir? 
we have NotFound.jsx file but I think it's not for all routes. it works for some routes.

**Comment by ciruz** [comment]
I answered something similar:

The current notFound state is more of a temporary solution. From the docs: https://beta.nextjs.org/docs/upgrade-guide#migrating-from-pages-to-app

> pages/404.js has been temporarily replaced with [notFound()](https://beta.nextjs.org/docs/api-reference/file-conventions/not-found) for specific route segments. We are also working on a new 404.js file to handle global errors.#

Also checkout the **App Directory Roadmap** https://beta.nextjs.org/docs/app-directory-roadmap 

![image](https://user-images.githubusercontent.com/987947/217282265-f27da48e-82ac-419e-b3f6-be4704cc2653.png)
